BadHabbit/BadHabbit/Kit/DesignKit/GlobalCoordinator/GlobalCoordinator.swift

```diff
@@ -83,9 +84,23 @@ final class GlobalCoordinator: IGlobalCoordinator {
             
         case let navVC as UINavigationController:
             return topViewController(on: navVC.topViewController)
-            
+
         default:
             return viewController
         }
     }
+
+    func pushVC(_ viewController: UIViewController) {
+        guard let window = self.window else {
+            assertionFailure()
+            return
+        }
+
+        guard let topVC = topViewController(on: window.rootViewController) else {
+            assertionFailure()
+            return
+        }
```

Comment:
To avoid code duplication we need to add function that returns top UIViewController and refactoring push and present functions

Reply:
resolved